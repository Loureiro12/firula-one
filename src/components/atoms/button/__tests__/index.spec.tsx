import { render } from "@testing-library/react-native";
import { Button } from "..";
import type { IButtonProps } from "../types";

jest.mock("@expo/vector-icons", () => ({
  Ionicons: "Ionicons",
}));

describe("Button Component", () => {
  const mockProps: IButtonProps = {
    label: "Click me",
    iconName: "accessibility",
    testIDPrefix: "login",
    onPress: jest.fn(),
  };

  test("renders label correctly", () => {
    const { getByText } = render(<Button {...mockProps} />);
    expect(getByText("Click me")).toBeTruthy();
  });

  test("renders icon correctly", () => {
    const { getByTestId } = render(<Button {...mockProps} />);
    expect(getByTestId("login-button")).toBeTruthy();
  });

  test("applies correct testID to Pressable", () => {
    const { getByTestId } = render(<Button {...mockProps} />);
    const button = getByTestId("login-button");
    expect(button).toBeTruthy();
  });

  test("matches snapshot", () => {
    const tree = render(<Button {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly without testIDPrefix", () => {
    const propsWithoutPrefix: IButtonProps = {
      label: "Click me",
      iconName: "home",
      onPress: jest.fn(),
    };
    const { getByText } = render(<Button {...propsWithoutPrefix} />);
    expect(getByText("Click me")).toBeTruthy();
  });
});
