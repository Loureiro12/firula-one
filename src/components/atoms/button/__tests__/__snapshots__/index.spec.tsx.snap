// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1FBD58",
      "borderRadius": 8,
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "center",
      "paddingVertical": 14,
      "width": "100%",
    }
  }
  testID="login-button"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "white",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    Ó®Å
  </Text>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Click me
  </Text>
</View>
`;
